.github/workflows/reusable-build.yml

```diff
@@ -0,0 +1,97 @@
+name: 🔒Reusable Workflow Build for Tests
+
+on:
+  workflow_call:
+    inputs:
+      image:
+        required: true
+        type: string
+        description: 'The image to use'
+      architecture:
+        required: true
+        type: string
+        description: 'The architecture to use'
+      build_configuration:
+        required: true
+        type: string
+        description: 'The build configuration to use'
+
+env:
+  # Disable the .NET logo in the console output.
+  DOTNET_NOLOGO: true
+  # Disable the .NET first time experience to skip caching NuGet packages and speed up the build.
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+  # Disable sending .NET CLI telemetry to Microsoft.
+  DOTNET_CLI_TELEMETRY_OPTOUT: true
+
+jobs:
+  get-version:
+    name: Calculating Version Suffix
+    runs-on: ubuntu-latest
+    outputs:
+      version_suffix: ${{ steps.set-vars.outputs.version_suffix }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - id: git-vars
+        name: Get git branch information
+        shell: bash
+        run: |
+          echo "##[set-output name=git_branch;]$(echo $GITHUB_REF)"
+          echo "::set-output name=git_hash::$(git rev-parse --short HEAD)"
+
+      - id: set-vars
+        uses: actions/github-script@v7
+        with:
+          script: |
+            let gitHash = "${{ steps.git-vars.outputs.git_hash }}";
+            let runNumber = "${{ github.run_number }}"; 
+            let rawGitRef = "${{ steps.git-vars.outputs.git_branch }}";
+            console.log("rawGitRef: " + rawGitRef);
+            let gitRef = rawGitRef.replace(/^refs\/heads\//, "");
+            if(gitRef.indexOf("refs/pull/") === 0) {
+              gitRef = "pr-" + gitRef.substring(10, gitRef.lastIndexOf("/"));
+            }
+            var versSuffix = `${gitRef}.${runNumber}+${gitHash}`;
```

Comment:
Will this go fine if the branch has the form `feature/xyz`?

Reply:
Looks like there are plentiful of special characters we need to replace. I've added support of replacing `/`, `!`, `@`, `#`, `$`, `%`, `&` with `_`